core/src/runtime/mod.rs

```diff
@@ -134,6 +141,8 @@ pub enum ExecutionError {
     Unimplemented(),
 }
 
+const NUM_PREALLOCATED_RANDOM_NUMS: usize = 1_000_000;
```

Comment:
constants should almost always be at the top of the file. this is a weird place to put a const